tensorboard/plugins/core/core_plugin_test.py

```diff
@@ -432,7 +461,20 @@ def _generate_test_data(self, run_name, experiment_name):
             else:
                 writer.add_graph(graph=None, graph_def=graph_def)
 
-        # Write data for the run to the database.
+        if ENABLE_DB_TESTS:
+            self._generate_test_data_db(run_name, experiment_name)
+
+        with tf.compat.v1.Session() as sess:
+            sess.run(tf.compat.v1.global_variables_initializer())
```

Comment:
Move this inside the `if HAS_TF_CONTRIB` block? It doesn’t do anything
otherwise, which is a bit confusing.

Once you’ve done that, you can just move this whole block in:

```python
        if ENABLE_DB_TESTS:
            self._generate_test_data_db(run_name, experiment_name)
            with tf.compat.v1.Session() as sess:
                sess.run(tf.compat.v1.global_variables_initializer())
                sess.run(tf.contrib.summary.summary_writer_initializer_op())
                sess.run(tf.contrib.summary.all_summary_ops())
```

…and then we don’t need `HAS_TF_CONTRIB` at all (can inline it into the
definition of `ENABLE_DB_TESTS`).

Reply:
Done.